lib/ansible/galaxy/token.py

```diff
@@ -35,6 +47,110 @@
 
 display = Display()
 
+AH_TOKEN_EXPIRES_N_MIN_EARLY = 1
+AH_AUTH_URL = "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token"
+AH_JWKS_URI = "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/certs"
+
+# https://www.rfc-editor.org/rfc/rfc7518#section-3.1
+REQUIRED_JWT_HEADER_KEYS = {'alg', 'typ', 'kid'}
+REQUIRED_JWT_HEADER_VALUES = {
+    'alg': ['RS256'],
+    'typ': ['JWT'],
+}
+
+
+def assert_jwt_structure(raw_jwt_string):
+    """Test that given JWT string looks valid.
+
+    :raises InvalidSignature: If it isn't.
+    """
+    if raw_jwt_string.count('.') != 2:
+        raise InvalidSignature(f"unsupported format: {raw_jwt_string}")
+
+    if any(
+        possible_whitespace in string.whitespace
+        for possible_whitespace in raw_jwt_string.rsplit('.')[0]
+    ):
+        raise InvalidSignature(f"JWS header and payload should not contain line breaks or whitespace: {raw_jwt_string.rsplit('.')[0]}")
+
+
+def assert_jwt_header(header):
+    """Test that given JWT header looks valid.
+
+    :raises InvalidSignature: If it isn't.
+    """
+    if (
+        set(header) != REQUIRED_JWT_HEADER_KEYS
+        or any(header[key] not in value for key, value in REQUIRED_JWT_HEADER_VALUES.items())
+    ):
+        raise InvalidSignature(f"unsupported header: {header}")
+
+
+def b64urlsafe_to_long(data):
+    decoded_bytes = base64.urlsafe_b64decode(
+        to_bytes(data, errors="surrogate_or_strict") + b"=="
+    )
+    return int(decoded_bytes.hex(), 16)
+
+
+def b64urlsafe_to_json(data):
```

Comment:
```suggestion
def b64urlsafe_to_dict(data):
```
Reading this, I don't think it returns a JSON string tho.

Reply:
Or
```suggestion
def parse_b64urlsafe_as_json(data):
```